apps/ciphers.c

```diff
@@ -81,6 +81,12 @@ static unsigned int dummy_psk(SSL *ssl, const char *hint, char *identity,
                               unsigned char *psk,
                               unsigned int max_psk_len)
 {
+    (void)ssl;
```

Comment:
Neither these.

Reply:
I think this one is too noisy, agreed.

Reply:
These are for -Wunused-parameter. There are thousands of such silenced unused params, which are hereby at least documented.

Reply:
I'm saying that silencing it is not particularly useful here -- you're making it harder to then discover where you want actual refactoring to drop the unused parameters. This is like turning off the smoke detector to drown out the noise.

We go from "here's a nice list of things where there's unused parameters we should fix" to "the warning is clean but we know we have tonnes of suppressions everywhere which are hard to grep for".

You'd be better off passing -Wno-unused-parameter, as you're not then hiding the problem.

Reply:
You got it reverse. 
* With  -Wno-unused-parameter you are silencing all such errors, and have no idea which params are ignored.
* With (void)param you enable the warnings, detect new unused params and if the param is used later, you get an error also.